dev-docs/design/how-lit-html-works.md

```diff
@@ -0,0 +1,431 @@
+
+#### Create the `TemplatePart`s
+
+A `TemplatePart` is metadata storing where a dynamic JavaScript value should be set on the DOM. Each `TemplatePart` contains the depth-first index of the node it is associated with, along with the type of expression (`'text'` or `'attribute'`) and the name of the attribute.
+
+`TemplatePart`s are found by walking the tree of nodes in the `<template>` element and finding the marker annotations. If a marker is found, either in an attribute, as a marker comment node, or in text content, a `TemplatePart` is created and stored in the `parts` class field.
+
+When traversing the tree of nodes, the following work is done for the different Node types:
+
+##### Element
+
+For each attribute on the element, if it has a `$lit$` suffix, then the expression represents either a `PropertyPart`, `BooleanAttributePart`, `EventPart`, or `AttributePart`. These can be distinguished by the first character in the attribute name ([see Appendix Parts table](#parts)).
+
+An attribute on the element may also represent an `ElementPart` in the case where the attribute name starts with the marker sentinel value. E.g. `<div ${} ${}>` is marked up as `<div lit$random$0="" lit$random$1="">`.
+
+For each of the bound attributes identified on the element, they are removed from the element and a `TemplatePart` is created to record the location, name, and type of dynamic binding.
+
+##### Comment
+
+A Comment is usually either an expression marker, or a user-written comment with no expression associated. Occasionally though a user may have written an expression inside a comment. We might see a comment like `<!--<div>${text}</div>-->` in the template text, so we must scan comments for marker text.
+
+A comment that matches a marker annotation is a `TemplatePart` in ChildPart position.
+
+##### Text
+
+Usually text-position markers will be comment nodes, but inside `<style>`, `<script>`, `<textarea>`, `<title>` tags the markup that _looks_ like a comment (`<!-- lit -->`) will be inserted as _text_. So we must search for the marker within Text nodes. If found we split the Text node and insert a comment marker, and store a `TemplatePart` in ChildPart position referencing the comment marker's node index.
+
+### 2.ii. Create
+
+The create phase is performed only when rendering for **the first time to a specific Part**.
+
+In the `lit-html` source code this phase is the [instantiating and cloning](https://github.com/lit/lit/blob/64fb960246e8f1eae982c2f09fca9759001756af/packages/lit-html/src/lit-html.ts#L1534-L1535) of a `TemplateInstance`.
+
+For the example, this phase results in the DOM nodes being cloned into a `fragment`, ready to be inserted in the document, and a list of `Parts` that can be updated by the next phase to set the dynamic JavaScript values in that fragment.
+
+```js
+const instance = new TemplateInstance(preparedTemplate);
+const fragment = instance._clone();
+
+console.log(instance._$parts);
```

Comment:
Can't think of a specific change to make here but I thought it weird that the code is creating `fragment` but the logging is actually accessing `instance`. `instance._clone()` actually has a side-effect of adding to `instance._$parts`.

Reply:
Added a comment about line 269.